kratos/mpi/utilities/parallel_fill_communicator.cpp

```diff
@@ -0,0 +1,466 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   Michael Andre, https://github.com/msandre
+//
+
+#include "parallel_fill_communicator.h"
+
+#include "includes/model_part.h"
+#include "includes/data_communicator.h"
+#include "processes/graph_coloring_process.h"
+#include "mpi/includes/mpi_communicator.h"
+
+namespace Kratos
+{
+
+ParallelFillCommunicator::ParallelFillCommunicator(ModelPart& r_model_part)
+    : mrBaseModelPart(r_model_part)
+{}
+
+void ParallelFillCommunicator::Execute()
+{
+    KRATOS_TRY
+    ComputeCommunicationPlan(mrBaseModelPart);
+    KRATOS_CATCH("");
+}
+
+void ParallelFillCommunicator::PrintDebugInfo()
+{
+    PrintModelPartDebugInfo(mrBaseModelPart);
+}
+
+void ParallelFillCommunicator::PrintModelPartDebugInfo(ModelPart& rModelPart)
```

Comment:
I would say this fct should accept a const ModelPart
```suggestion
void ParallelFillCommunicator::PrintModelPartDebugInfo(const ModelPart& rModelPart)
```
If that doesn't work then there is sth fishy going on :)

Reply:
Also this fct uses cout and the old error macro
Not sure though if it worth to be updated